### PR TITLE
Revise evaluation code

### PR DESCRIPTION
**Pull request by molereddy**
Curent evaluation configs and code does not distinguish between data splits.
- [x] restructure eval dump format 
- [x] rename yaml files for forget, update metrics and memorization files to fit new names
- [x] remove truth ratio dependence on paraphrased (access key)
- [x] fix truth ratio implementation to match TOFU
- [ ] create yaml files for other splits' metrics
- [ ] Implement metric aggregator for every metric {"overall":0.5, "details": {"index_to_probs": {}}}
- [ ] write truth ratio version for each split with averaging logic for different datasets in Truth Ratio

### CODE REVIEW

File: PR-todos.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- [x] rename yaml files for forget
```
Comment (Dornavineeth) [inline PR-todos.md:1]: This file should be removed

File: configs/eval/tofu.yaml [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
-    - Truth_Ratio
+    - forget_Q_A_Prob
+    # - forget_Q_PARA_A_PARA_Prob
+    # - forget_Q_A_PERT_Prob
+    # - forget_Q_A_ROUGE
+    # - forget_Q_PARA_A_PARA_ROUGE
+    - forget_Q_A_PARA_ROUGE
+    - forget_Q_A_PERT_ROUGE
+    - full_BIO_Prob
+    - full_BIO_ROUGE
```
Comment (Dornavineeth) [inline configs/eval/tofu.yaml:15]: Shall we comment all the BIO metrics?
Also Lets include metrics which are only inlcuded in the TOFU.

File: src/evals/metrics/memorization.py [hunk 3]
```diff
@@ -21,27 +23,43 @@ def evaluate_probability_batch(model, batch):
...
-                        index_to_prob[idx].append({"prob": prob})
-                    else:
-                        index_to_prob[idx] = [index_to_prob[idx]] + [{"prob": prob}]
-                else:
-                    index_to_prob[idx] = {"prob": prob}
-    return index_to_prob
+        for intra_item_idx, mini_batch in batch.items():
+            data_indices = mini_batch.pop("index").cpu().numpy().tolist() # data item indices
+            batch_evals = evaluate_probability_batch(model, mini_batch) # batch_evals maps each attr to a list of length bsz
+            transpose_batch_evals = [  # a list of len bsz which maps for each attr to an indiv value
```
Comment (Dornavineeth) [inline src/evals/metrics/memorization.py:42]: Lets **not** have the comment inside the list. Better keep it above the line. 
Simple comment dict(str: list)->list(dict(str: float)) would be easy to understand.

File: src/evals/metrics/memorization.py [hunk 4]
```diff
@@ -21,27 +23,43 @@ def evaluate_probability_batch(model, batch):
     num_token_gt = (batch["labels"] != -100).sum(-1)
     avg_loss = loss / num_token_gt
     normalized_probs = torch.exp(-avg_loss)
-    return normalized_probs.cpu().numpy().tolist()
+    return {'prob': normalized_probs.cpu().numpy().tolist(), 'avg_loss': avg_loss.cpu().numpy().tolist()}
 
 
 def evaluate_probability(model, dataloader):
```
Comment (Dornavineeth) [inline src/evals/metrics/memorization.py:29]: Same changes should be also applied to evaluate_similarity function in this file.

Comment (Dornavineeth) [inline src/evals/metrics/memorization.py:29]: Same changes should be also applied to evaluate_similarity function in this file.

File: src/evals/metrics/memorization.py [hunk 5]
```diff
@@ -128,20 +146,27 @@ def rouge(model, **kwargs):
...
-    return index_to_scores
+    return scores_by_index
+
+
+@unlearning_metric(name="forget_truth_ratio")
+def forget_truth_ratio(model, **kwargs):
+    def aggregate(x):
+        return sum(x) / len(x) if isinstance(x, list) else x
+    correct_answer_results = kwargs["pre_compute"]["paraphrase"]
+    correct_loss = {idx: aggregate(result["avg_loss"])
```
Comment (Dornavineeth) [inline src/evals/metrics/memorization.py:160]: Why do we need to store `avg_loss`? 
Cant we just take GM of the probability and skip returning avg_loss?

File: src/evals/metrics/memorization.py (hunk 4 above)
Comment (molereddy) [reply src/evals/metrics/memorization.py]: I refactored code fully now check all again
